### DISCUSSION TITLE
Spring 6+ upgrades trigger migration to HttpClient 5 – how to deal with undeclared dependency on HttpClient 4?

### DISCUSSION THREAD

**Discussion by DidierLoiseau** [openrewrite/rewrite#4584] [Q&A] [answered]
I noticed an issue with this migration, which is triggered transitively by Spring Boot 3+ upgrades since openrewrite/rewrite-spring#566: HttpClient 4 is a relatively common dependency, so people often use it without even realizing it is pulled transitively by another dependency, and they don’t declare HttpClient as an explicit dependency.

The problem with this migration is that it will update the code without adding the HttpClient 5 dependency if HttpClient 4 was missing.

Is there a way around this?

**Comment by timtebeek** [comment, marked as answer]
Thanks for reporting this oversight; I've logged an issue with the likely steps to resolve this; hope that helps!
- https://github.com/openrewrite/rewrite-apache/issues/35
